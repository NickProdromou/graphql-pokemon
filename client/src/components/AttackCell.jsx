import React from "react";
import PropTypes from "prop-types";
import Styled from "styled-components";

import PokemonTypeTag from "./PokemonTypeTag";
import { type as typeMixin } from "../style/mixins/index";
import { spacing, colours } from "../style/variables";

const AttackCell = ({ name, type, damage, category }) => (
  <StyledAttack>
    <h3 className="AttackTitle">{name}</h3>    
    <div className="AttackDamage">
     damage: {damage}
    </div>
    <div className="AttackCategory">
    category: {category}
    </div>
    <div className="AttackType">
      <PokemonTypeTag type={type} />
    </div>
  </StyledAttack>
);

AttackCell.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  damage: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired
};

const StyledAttack = Styled.div`
  padding: ${spacing.small.level5} ${spacing.small.level2};
  background: ${colours.borderColor};
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  :not(:last-child) {
    margin-bottom: ${spacing.small.level4};
  }

  .AttackTitle {
    ${typeMixin("heading3")}
    margin-bottom: ${spacing.small.level3};
    margin-top: 0;
  }

  .AttackDamage {
    ${typeMixin("detail")}
    margin-bottom: ${spacing.small.level2};
  }

  .AttackCategory {
    ${typeMixin("detail")}
  }

  .AttackType{
    margin-top: ${spacing.small.level5};
  }
`;

export default AttackCell;
