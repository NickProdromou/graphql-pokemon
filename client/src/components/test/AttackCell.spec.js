import React from "react";
import renderer from "react-test-renderer";
import AttackCell from "../AttackCell";

describe("AttackCell component", () => {
  const props = {
    type: "Fire"
  };

  test("renders", () => {
    const props = {
      name: "foo",
      type: "bar",
      damage: 54,
      category: "baz"
    };

    const component = renderer.create(
      <AttackCell
        name={props.name}
        type={props.type}
        damage={props.damage}
        category={props.category}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
