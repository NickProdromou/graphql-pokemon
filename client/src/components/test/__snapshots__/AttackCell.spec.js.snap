// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttackCell component renders 1`] = `
<div
  className="sc-bwzfXH jCIkPP"
>
  <h3
    className="AttackTitle"
  >
    foo
  </h3>
  <div
    className="AttackDamage"
  >
    damage: 
    54
  </div>
  <div
    className="AttackCategory"
  >
    category: 
    baz
  </div>
  <div
    className="AttackType"
  >
    <span
      className="TypeTag sc-bdVaJa eBMyvA"
      type="bar"
    >
      bar
    </span>
  </div>
</div>
`;
